sklearn/ensemble/_weight_boosting.py

```diff
@@ -1078,9 +1078,29 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Fitted AdaBoostRegressor estimator.
         """
+        # Validate scalar parameters
+        check_scalar(
+            self.n_estimators,
+            "n_estimators",
+            target_type=numbers.Integral,
+            min_val=1,
+            include_boundaries="left",
+        )
+
+        check_scalar(
+            self.learning_rate,
+            "learning_rate",
+            target_type=numbers.Real,
+            min_val=0,
+            include_boundaries="neither",
+        )
```

Comment:
We can move both of these checks into `BaseWeightBoosting`, and replace the check for `self.learning_rate`:

https://github.com/scikit-learn/scikit-learn/blob/c9e5067cb14de578ab48b64f399743b994e3ca94/sklearn/ensemble/_weight_boosting.py#L115-L116

Reply:
Thank you. In that case I will remove the checks I added to [`AdaBoostClassifier`]( #21442) too. They are the exact same checks as what I am about to move into `BaseWeightBoosting`.